### DISCUSSION TITLE
White label information

### DISCUSSION THREAD

**Discussion by owangen** [Koenkk/zigbee2mqtt#8584] [General]
Hi
I have a generic question about how we should mark white label information. Should white label information be documented both places (both at the "original" product and the "white labeled" product)? 
Right now it's not consistent documented, and I want to clean it up.

**Comment by Koenkk** [comment]
It's indeed documented at both, whitelabeled devices don't have product pages on their own and therefore refer to the "original" devices.

Note that this is generated from the `whitelabel` property (e.g. https://github.com/Koenkk/zigbee-herdsman-converters/blob/eb7bc618fb2238b58d86ec06f6e621fcf42f713b/devices/tuya.js#L21 )

**Reply by owangen** [reply]
Ok, thanks. Yes, I guessed this information was extracted from converters code base. I noticed that some Sunricher devices is wrongly pointing to Namron devices, and I'm unsure what decides what becomes the "original" device if white label information is documented both places.

**Reply by owangen** [reply]
@Koenkk After reading the document generation code I see what the problem is.
The problem is that there has been some misunderstanding about how the white label information works in zigbee-herdsman-converters. For example with Sunrichers devices. Many of the devices from Robb, Iluminize and Namron (maybe more) are white label products of Sunricher. So Sunricher should be the "original" device, but right now it's mixed up in the zigbee-herdsman-converters codebase. I'll propose a PR to fix it later

**Reply by Koenkk** [reply]
How do you know Sunricher is the "original" vendor, what is shown for the `manufacturerName` in the `database.db`?

**Reply by owangen** [reply]
I have one Namron device I'm testing with, its model number is 4512726 and in database.db it says `"manufacturerName":"NAMRON AS"`
The most clear evidence that Sunricher is the "original" vendor at least for (most) Namron products are when I download the user manual for this device from this page. https://www.namron.com/products/namron-zigbee-4-i-1-vridimmer/#tabs-3 (called "Brukerveiledning") The user manual https://www.elektroimportoren.no/docs/lib/4512726-Brukerveiledning-5.pdf clearly states "SR-ZG2835" in the header. Same thing with same Robb device https://www.robbshop.nl/detail/c7711f6416fb45b8bb3c89da37f0d585 when I download the user manual it says "SR-ZG2835" in the header.

**Reply by Koenkk** [reply]
Have you verified that the sunricher variant of this device uses the same `zigbeeModel`?

**Reply by owangen** [reply]
I don't have any Sunricher or Robb devices to test with, but it looks that they have different zigbeeModel from what I read here https://github.com/dresden-elektronik/deconz-rest-plugin/issues/3085.

**Reply by Koenkk** [reply]
If they have a different Zigbee model they shouldn't be marked as whitelabel, this is only done when devices share the same zigbeeModel and thus cannot be distinguished from each other.

**Reply by owangen** [reply]
I see... Then I misunderstood the whitelabel marking concept. I'll fix the Namron devices whitelabel markings (since I added them in the first place).

**Reply by owangen** [reply]
I have submitted a PR fixing this now.
Maybe I'm just nitpicking, but since Sunricher is the "original" white label vendor I suspect the following to be wrong (meaning the white label information should be marked on the Sunricher device):
https://github.com/Koenkk/zigbee-herdsman-converters/blob/247d3dcd745d07244cde931b51c4b3a639e0fdbe/devices/robb.js#L81
https://github.com/Koenkk/zigbee-herdsman-converters/blob/247d3dcd745d07244cde931b51c4b3a639e0fdbe/devices/robb.js#L94
https://github.com/Koenkk/zigbee-herdsman-converters/blob/247d3dcd745d07244cde931b51c4b3a639e0fdbe/devices/robb.js#L106
https://github.com/Koenkk/zigbee-herdsman-converters/blob/247d3dcd745d07244cde931b51c4b3a639e0fdbe/devices/robb.js#L131
https://github.com/Koenkk/zigbee-herdsman-converters/blob/cf3dcc3cc252325728b24909d5f2b3dd7a54aa1d/devices/icasa.js#L55
https://github.com/Koenkk/zigbee-herdsman-converters/blob/58d429436d74f287507a5caa148d524ef930c480/devices/nue_3a.js#L141
https://github.com/Koenkk/zigbee-herdsman-converters/blob/a0c21014ecf7e9115cfe600c13d26379ffa2e8e7/devices/iluminize.js#L131

**Reply by Koenkk** [reply]
I don't think that e.g. https://github.com/Koenkk/zigbee-herdsman-converters/blob/247d3dcd745d07244cde931b51c4b3a639e0fdbe/devices/robb.js#L81 should be a whitelabel at all and a seperate definition can be created in sunricher.js.

https://github.com/Koenkk/zigbee-herdsman-converters/blob/247d3dcd745d07244cde931b51c4b3a639e0fdbe/devices/robb.js#L69, likely:
- ZG2833K8_EU05 = sunricher SR-ZG9001K8-DIM
- ROB_200-007-0 = robb ROB_200-007-0 (so correct already)

**Reply by owangen** [reply]
Hmm, I'm not to comfortable changing it without a possibility to test it...
